src/core_plugins/timelion/server/lib/offset_time.js

```diff
@@ -34,3 +34,37 @@ export default function offsetTime(milliseconds, offset, reverse) {
   const momentObj = moment(milliseconds)[mode](parts[1], parts[2]);
   return momentObj.valueOf();
 }
+
+function timeRangeErrorMsg(offset) {
+  return `Malformed timerange offset, expecting "timerange:<number>", received: ${offset}`;
+}
+
+/*
+ * Calculate offset when parameter is requesting a relative offset based on requested time range.
+ *
+ * @param {string} offset - offset parameter value
+ * @param {number} from - kibana global time 'from' in milliseconds
+ * @param {number} to - kibana global time 'to' in milliseconds
+ */
+export function preprocessOffset(offset, from, to) {
+  if (!offset.startsWith('timerange')) {
+    return offset;
+  }
+
+  const parts = offset.split(':');
+  if (parts.length === 1) {
+    throw new Error(timeRangeErrorMsg(offset));
+  }
+
+  const factor = parseFloat(parts[1]);
+  if (isNaN(factor)) {
+    throw new Error(timeRangeErrorMsg(offset));
+  }
+  if (factor === 0) {
```

Comment:
should we allow to set offset to 0 (which would mean no offset) ? think it would make sense.

Reply:
I don't think setting the offset to zero makes sense. Currently, you can not set an offset of zero

<img width="2045" alt="screen shot 2018-06-14 at 4 00 49 pm" src="https://user-images.githubusercontent.com/373691/41440583-4be621ca-6fec-11e8-8926-c10af08651b9.png">